activerecord-oracle_enhanced-adapter.gemspec

```diff
@@ -26,4 +26,9 @@ This adapter is superset of original ActiveRecord Oracle adapter.
   s.test_files = Dir["spec/**/*"]
   s.add_runtime_dependency("activerecord", ["~> 7.1.0.alpha"])
   s.add_runtime_dependency("ruby-plsql", [">= 0.6.0"])
+  if /java/.match?(RUBY_PLATFORM)
+    s.platform = Gem::Platform.new("java")
+  else
+    s.add_runtime_dependency("ruby-oci8")
+  end
```

Comment:
AFAIK, this branch in gemspec depends on the release environment. Not the user's runtime environment.
I'm not sure, will Oracle enhanced adapter release two types (MRI and JRuby) in future? I'm not familiar with it, the release process of Oracle enhanced adapter may be (a bit?) complicated.

Reply:
I think we need to push two gems one is built with CRuby and another one is built with JRuby as suggested at https://github.com/rails/rails/issues/44108#issuecomment-1007554627 to avoid -d oracle option will be removed from Rails.

Reply:
Until JRuby supports Ruby 3.0 https://github.com/jruby/jruby/issues/6878, only one gem file will be pushed to ruby gems.org .

Reply:
The new release process is that running `rake release` (or `gem push`) in each of MRI and JRuby environments, right?

Reply:
> Until JRuby supports Ruby 3.0 jruby/jruby#6878, only one gem file will be pushed to ruby gems.org .

Rails 7.0 supports Ruby 2.7. Do the Oracle enhanced adapter gem need to release for JRuby?
https://rubygems.org/gems/rails/versions/7.0.1

Reply:
> The new release process is that running rake release (or gem push) in each of MRI and JRuby environments, right?

Yes.

Reply:
I get it. I'm not sure because I haven't released it for each platform, but it looks good to me.

Reply:
> Rails 7.0 supports Ruby 2.7. Isn't it necessary to release to JRuby?

It is not necessary now becaue JRuby skips supporting CRuby 2.7 and it is going to support CRuby 3.0. Then once JRuby supports CRuby 3.0, we will release Oracle enhanced adapter gem for JRuby and CRuby. but not yet.

Reply:
I overlooked the important part of the link. I understand the context. Thank you!